File: Gotho.BlazorPdf/Pdf/PdfState.cs
namespace Gotho.BlazorPdf.Pdf;

internal class PdfState
{
    public required string? Id { get; set; }
    public required string? Url { get; set; }
    public required string Source { get; set; }

    public required int CurrentPage { get; set; }
    public required int Orientation { get; set; }
    public required double Scale { get; set; }

    public required string? Password { get; set; }

    public required bool DrawLayerEnabled { get; set; }
    public required string PenColor { get; set; }
    public required int PenThickness { get; set; }

    public required string? SearchQuery { get; set; }
    public required int? ActiveResultIndex { get; set; }
}

File: Gotho.BlazorPdf/Ts/PdfState.ts
export class PdfState {
    public id: string | null = null;
    public url: string | null = null;
    public orientation: number = 0;
    public scale: number = 1;
    public currentPage: number = 1;
    public singlePageMode: boolean = true;
    public password: string | null = null;
    public source: string = "Url";
    public drawLayerEnabled: boolean = false;
    public penColor: string = "#000000";
    public penThickness: number = 1
}